File: task3-python/attendant.py
# CSCI3180 Principles of Programming Languages
#
# --- Declaration ---
#
# I declare that the assignment here submitted is original except for source
# material explicitly acknowledged. I also acknowledge that I am aware of
# University policy and regulations on honesty in academic work, and of the
# disciplinary guidelines and procedures applicable to breaches of such policy
# and regulations, as contained in the website
# http://www.cuhk.edu.hk/policy/academichonesty/
#
# Assignment 2
# Name : Ling Leong
# Student ID : 1155062557
# Email Addr : alanalan0124@yahoo.com.hk

class Attendant:
    def __init__(self, name, weight, wealth, position):
        self._position = position
        self._wealth = wealth
        self._weight = weight
        self._name = name

    def move_to(self, destination, obj=None):
        if obj is None:
            distance = abs(self._position - destination.position)
            calories_consumed = distance * self._weight
            self._position = destination.position
            print self._name + " moved to " + destination.name + "."
            return calories_consumed
        else:
            distance = abs(obj.position - destination.position)
            calories_consumed = distance * obj.weight
            obj.position = destination.position
            print obj.name + " was moved to " + destination.name + "."
            return calories_consumed

    def carry_to(self, obj, destination):
        calories_consumed = 0
        calories_consumed += self.move_to(obj)
        calories_consumed += self.move_to(destination)
        calories_consumed += self.move_to(destination, obj)
        print self._name + " consumed " + "%.1f" % calories_consumed + " calories."

    @property
    def position(self):
        return self._position

    @property
    def wealth(self):
        return self._wealth

    @property
    def weight(self):
        return self._weight

    @property
    def name(self):
        return self._name

    def pay(self, amount):
        if amount <= self._wealth:
            self._wealth -= amount
            print self._name + " paid " + "%.1f" % amount + " dollars."
            return amount
        else:
            return -1

File: task4-java/src/assg2task4/Attendant.java
/*
 *  CSCI3180 Principles of Programming Languages
 *
 *  --- Declaration ---
 *
 *  I declare that the assignment here submitted is original except for source
 *  material explicitly acknowledged. I also acknowledge that I am aware of
 *  University policy and regulations on honesty in academic work, and of the
 *  disciplinary guidelines and procedures applicable to breaches of such policy
 *  and regulations, as contained in the website
 *  http://www.cuhk.edu.hk/policy/academichonesty/
 *
 *  Assignment 2
 *  Name : Ling Leong
 *  Student ID : 1155062557
 *  Email Addr : alanalan0124@yahoo.com.hk
 */

package assg2task4;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author kenchan
 */
public class Attendant {
    protected String name;
    protected double weight;
    protected double wealth;
    protected int position;
    public Attendant(String Name, double Weight, double Wealth, int Position){
        this.name = Name;
        this.weight = Weight;
        this.wealth = Wealth;
        this.position = Position;
    }
    public double MoveTo(Room Destination){
        int distance = Math.abs(this.position - Destination.GetPosition());
        double calories_consumed = distance * this.weight;
        this.position = Destination.GetPosition();
        System.out.println(this.name + " moved to " + Destination.GetName() + "." );
        return calories_consumed;
    }
    public double MoveTo(Fixture Destination){
        int distance = Math.abs(this.position - Destination.GetPosition());
        double calories_consumed = distance * this.weight;
        this.position = Destination.GetPosition();
        System.out.println(this.name + " moved to " + Destination.GetName() + "." );
        return calories_consumed;
    }
    public double MoveTo(Room Destination, Fixture Obj){
        int distance = Math.abs(Obj.GetPosition() - Destination.GetPosition());
        double calories_consumed = distance * Obj.GetWeight();
        Obj.position = Destination.GetPosition();
        System.out.println(Obj.name + " was moved to " + Destination.GetName() + "." );
        return calories_consumed;
    }
    public void CarryTo(Fixture Obj, Room Destination){
        double calories_consumed = 0;
        //Move the attendant to the object first
        calories_consumed += this.MoveTo(Obj);
        //Move the attendant to the destination
        calories_consumed += this.MoveTo(Destination);
        //Move the object to the destination 
        calories_consumed += this.MoveTo(Destination, Obj);
        //Print calories consumed
        System.out.println(this.name + " consumed " + calories_consumed + " calories." );
    }
    public String GetName(){
        return this.name;
    }
    public int GetPosition(){
        return this.position;
    }
    public double GetWeight(){
        return this.weight;
    }
    public double Pay(double Amount){
        if (Amount <= this.wealth){
            this.wealth -= Amount;
            System.out.println(this.name + " paid " + Amount + " dollars.");
            return Amount;
        }
        else{
            return -1;
        }
    }
}
